src/Core/model/TaskStorage.cpp

```diff
@@ -10,10 +10,15 @@ bool TaskStorage::addTask(std::shared_ptr<FullTask> sft){
 }
 
 std::optional<std::weak_ptr<FullTask>> TaskStorage::getTask(TaskID id){
-    if (allTasks_.find(id) == allTasks_.end())
+    if (allTasks_.find(id) == allTasks_.end()) {
         return std::nullopt;
-    else
+    }
+    else {
+        if (!allTasks_[id]){
+            throw std::runtime_error("memory leak");
+        }
         return allTasks_[id];
```

Comment:
There are 3 find methods of allTasks 
1) allTasks_.find
2) else -> if (!allTasks_[id]
3) return
Why don't you store it once to variable and usind?

Reply:
fixed